### PR TITLE
{Sounds,Animations}Loader: lazy loading

### PR DESCRIPTION
**Pull request by MaxKellermann**
This reduces memory usage by 1.2 GB (tested on Sagas, sound off).

### CODE REVIEW

File: src/ClassicUO.Assets/SoundsLoader.cs [hunk 1]
```diff
@@ -53,190 +54,221 @@ public class SoundsLoader : UOFileLoader
...
+                _musicData.Add(65, new Tuple<string, bool>("serpentislecombat_u7", true));
+                _musicData.Add(66, new Tuple<string, bool>("valoriaships", true));
+            }
+        }
+
+        private void EnsureMusicLoaded()
+        {
+            musicMutex.WaitOne();
+            try {
+                if (!musicLoaded) {
```
Comment (kamronbatman) [inline src/ClassicUO.Assets/SoundsLoader.cs]: I think maybe we might want to do a volatile read or interlock compare before the mutex?

And I have used AutoResetEvent in these cases instead, not sure of the performance difference though.

Comment (MaxKellermann) [reply src/ClassicUO.Assets/SoundsLoader.cs]: Why?

Comment (kamronbatman) [reply src/ClassicUO.Assets/SoundsLoader.cs]: Let's take a step back. I thought CUO is single threaded and you removed the task for backgrounding the file load, which makes sense. So why do we need a Mutex at all?

Comment (MaxKellermann) [reply src/ClassicUO.Assets/SoundsLoader.cs]: Oh, we're completely single-threaded? If that is guaranteed (and no offloaded Task can ever access these data structures), of course we can easily remove the mutex.

**Comment by MaxKellermann** [comment]
Removed all mutexes.

**Review by kamronbatman — APPROVED** [review]

**Pull request merged**